### ISSUE TITLE
pytest-assume turns failed assertions into success on unittest.TestCase-based tests

### ISSUE THREAD

**Issue by antoche** [astraw38/pytest-assume#41]
The following case:

```python
import pytest

from unittest import TestCase

def test_bar():
    with pytest.assume:
        assert 1==2

class Foo(TestCase):
    def test_foo(self):
        with pytest.assume:
            assert 1==2

    def test_bar(self):
        assert 1==2                           
```

Results in this:

```
$ pytest test_assume.py 
============================================================ test session starts =============================================================
platform linux -- Python 3.7.3, pytest-4.6.5, py-1.8.0, pluggy-0.13.1 -- /vol/apps/python/3.7.3/bin/python
cachedir: .pytest_cache
rootdir: /tmp/pytest_assume
plugins: assume-2.3.2
collected 3 items                                                                                                                            

test_assume.py::test_bar FAILED                                                                                                        [ 33%]
test_assume.py::Foo::test_bar FAILED                                                                                                   [ 66%]
test_assume.py::Foo::test_foo PASSED                                                                                                   [100%]

================================================================== FAILURES ==================================================================
__________________________________________________________________ test_bar __________________________________________________________________

    def test_bar():
        with pytest.assume:
>           assert 1==2
E           pytest_assume.plugin.FailedAssumption: 
E           1 Failed Assumptions:
E           
E           test_assume.py:7: AssumptionFailure
E           >>	assert 1==2
E           AssertionError: assert 1 == 2
E             -1
E             +2

test_assume.py:7: FailedAssumption
________________________________________________________________ Foo.test_bar ________________________________________________________________

self = <test_assume.Foo testMethod=test_bar>

    def test_bar(self):
>       assert 1==2
E       AssertionError: assert 1 == 2
E         -1
E         +2

test_assume.py:15: AssertionError
===================================================== 2 failed, 1 passed in 0.04 seconds =====================================================
```

Note how `Foo::test_foo` is marked as passing when it really should not. This is unexpected and pretty dangerous.

**Comment by astraw38** [comment]
Some quick debugging shows that ``pytest_pyfunc_call`` hook is not being called at all for ``unittest.TestCase``-based tests. I'll dig further...

**Comment by astraw38** [comment]
fixed in #42

**Comment by antoche** [comment]
Nice, thanks for the quick fix!
